### ISSUE TITLE
Use of uninitialized value in tls1_enc could lead to crashes (but currently not reachable)

### ISSUE THREAD

**Issue by guidovranken** [openssl/openssl#2647]
This line: https://github.com/openssl/openssl/blob/master/ssl/record/ssl3_record.c#L894

assumes that ```n_recs``` >= 1.

I've investigated paths leading to this function and currently none will ever call it with ```n_recs``` == 0 (checked both master and 1.1.0.d) so at present it's not an issue, but it should be fixed.

**Comment by TheMassController** [comment]
I cant find your line anymore, was it accidentaly fixed?

**Comment by guidovranken** [comment]
No, if ```tls1_enc``` is called with ```nsize == 0``` (second argument), then

```c
        tmpr = EVP_Cipher(ds, recs[0].data, recs[0].input,
                          (unsigned int)reclen[0]);
```

accesses ```reclen[0]```, which is uninitialized. Possibly the other variables are uninitialized as well, but this one specifically showed up in my static analyzer.

**Comment by richsalz** [comment]
Fixed via #2822; please re-open if not fixed.
